packages/telestion-client-core/src/components/telestion-client.tsx

```diff
@@ -72,14 +73,28 @@ export function TelestionClient({
 	children
 }: TelestionClientProps) {
 	const update = usePreferences(selector);
+	const [target, targetSize] = useSize<HTMLDivElement>();
+	const [size, setSize] = useState<DOMRect>();
+
+	useEffect(() => {
+		if (targetSize) {
+			setSize(targetSize);
+		}
+	}, [targetSize]);
 
 	useEffect(() => {
 		update('null', 'title', title);
 	}, [title, update]);
 
 	// Fix Typescript bad types for JSX by wrapping components in Fragments to hide dumb types
 	// eslint-disable-next-line react/jsx-no-useless-fragment
-	return <>{wrapper ? wrapper(children) : children}</>;
+	return (
+		<div ref={target}>
+			<SizeContext size={size}>
+				{wrapper ? wrapper(children) : children}
+			</SizeContext>
+		</div>
+	);
```

Comment:
This will have to move to `telestion-client-common` (e.g., as part of the [`CommonWrapper`](https://wuespace.github.io/telestion-client/@wuespace/telestion-client-common/#CommonWrapper)) since `telestion-client-core` is meant to be environment-agnostic (e.g., also working with React Native).